app.js

```diff
@@ -69,108 +66,89 @@ class ChallengeCard {
     }
 
     return starContainer;
-  }
+  } */
 
-  createCard() {
-    const {
-      id,
-      title,
-      description,
-      rating,
-      minParticipants,
-      maxParticipants,
-      labels,
-      image,
-    } = this.challenge;
 
-    const card = document.createElement("div");
-    card.id = id;
-    card.classList.add("card");
-    card.classList.add("rooms__box");
-
-    const titleElement = document.createElement("h2");
-    titleElement.textContent = title;
-
-    const descriptionElement = document.createElement("p");
-    descriptionElement.textContent = description;
-
-    const detailsContainer = document.createElement("div");
-    detailsContainer.classList.add("rating-container");
-
-    const ratingElement = document.createElement("div");
-    ratingElement.classList.add("rating");
-    ratingElement.appendChild(this.generateStarImages(rating));
 
     const participantsElement = document.createElement("p");
-    participantsElement.textContent = `Participants: ${minParticipants} - ${maxParticipants}`;
+    participantsElement.classList.add("rooms__participants");
+    participantsElement.textContent = `${this.data.minParticipants} - ${this.data.maxParticipants} Participants`;
+
+    const typeOfRoom = document.createElement("p");
+    typeOfRoom.type = `Type: ${this.data.type}`;
 
     detailsContainer.appendChild(ratingElement);
     detailsContainer.appendChild(participantsElement);
-
-    const labelsElement = document.createElement("div");
-    labelsElement.classList.add("labels");
-    labels.forEach((label) => {
-      const labelElement = document.createElement("span");
-      labelElement.textContent = label;
-      labelsElement.appendChild(labelElement);
-    });
+    detailsContainer.appendChild(typeOfRoom);
 
     const imageElement = document.createElement("img");
-    imageElement.src = image;
-    imageElement.alt = title;
+    imageElement.src = this.data.image;
+    imageElement.alt = this.data.title;
     imageElement.classList.add("rooms__img");
 
+    const bookButton = document.createElement("button");
+    bookButton.textContent = "Book this room";
+    bookButton.classList.add("rooms__button");
+
     card.appendChild(imageElement);
     card.appendChild(titleElement);
-    card.appendChild(detailsContainer);
     card.appendChild(descriptionElement);
-    card.appendChild(labelsElement);
 
     return card;
   }
+}
 
-  getCardElement() {
-    return this.cardElement;
+// FETCH API
```

Comment:
This comment provides very little value to a reader of this code. Comments should only be included when the code they describe is difficult to understand, and the only way to explain it is using a comment. The first option should always be to change the code, especially names, but in this case, I don't think that's necessary.

The thing is that comments are often (mistakingly) left unchanged when code is changed, causing the comment to become obsolete and very misleading. Here's what I wrote about that same subject to a class mate of yours:

> My theory is that most programmers stop reading comments, unless they don't understand the code. The comments are colored differently and programmers, being lazy, just filter them out to have to read less. This makes sense, but it does mean that when you make changes to code, it's very easy to forget updating comments to reflect those changes.
> 
>For that reason, it's usually a good idea to avoid having comments, and instead, writing the code in a way that makes it easily readable, using good variable names etc.

In this case, the `class APIadapter` and it's method `fetchAllChallenges()` have good enough names to explain what this code is about, and the "FETCH API" comment does nothing to improve that. But it's very likely that someone in the future might move the `APIadapter` class, but forget to move the comment along, and then the code will become very confusing as the "FETCH API" comment will then be directly above `class ViewAll`.

That's why I always try to avoid comments unless it's the only way to make my code understandable (e.g. if the code is for a complex algorithm).